### ISSUE TITLE
[question] this works with @Async?

### ISSUE THREAD

**Issue by plentz** [spring-attic/spring-tenancy#4]
I've looked at the code and didn't found anything that made me believe it will work with @Async methods. Am I right?

**Comment by ghost** [comment]
@plentz the spring-tennancy framework does not currently have anything specific related to @Async.  I haven't looked at it closely, but it could just be a matter of implementing another TenancyContextHolderStrategy.

**Comment by rpraes** [comment]
@plentz I´m using spring-tenancy in my project . Did you solve this issue with @Async methods or have any workaround?

**Comment by plentz** [comment]
@rpraes not really. There's other points of this structure that doesn't fully cover a multi-tenancy application, so I really recommend you to test it before calling your app multi-tenant - you could have some unpleasant surprises.

**Comment by rpraes** [comment]
@plentz Yeah, i´m testing this architecture. There are some details that could be a problem in large applications, for example, changing only the schema you will not be able to use database cluster, but i think for small applications it can be used as starting.
